src/dxgi/dxgi_output.cpp

```diff
@@ -637,4 +621,51 @@ namespace dxvk {
     }
   }
 
+
+  void DxgiOutput::CacheMonitorData() {
+    // Try and find an existing monitor info.
+    {
+      DXGI_VK_MONITOR_DATA* pMonitorData;
+      if (SUCCEEDED(m_monitorInfo->AcquireMonitorData(m_monitor, &pMonitorData))) {
+        m_metadata = pMonitorData->DisplayMetadata;
+        return;
+      }
+    }
+
+    // Init monitor info ourselves.
+    // 
+    // If some other thread ends up beating us to it
+    // by another InitMonitorData, it doesn't really matter.
+    // 
+    // The only thing we cache from this is the m_metadata which
+    // should be exactly the same.
+    // We don't store any pointers from the DXGI_VK_MONITOR_DATA
+    // sturcture, etc.
+    DXGI_VK_MONITOR_DATA monitorData = {};
+
+    // Query current display mode
+    wsi::WsiMode activeWsiMode = { };
+    wsi::getCurrentDisplayMode(m_monitor, &activeWsiMode);
+
+    // Get the display metadata + colorimetry
+    wsi::WsiEdidData edidData = wsi::getMonitorEdid(m_monitor);
+    if (!edidData.empty()) {
```

Comment:
should probably issue a warning if we can't find the EDID too

Reply:
wsi::getMonitorEdid already does this.

Reply:
Done.